test/deterministic/one_run_deterministic.jl

```diff
@@ -1,34 +1,26 @@
 
 @testset "run deterministic" begin
     T = 3
-    dir = @__DIR__
-
     parameters = BeforeIT.AUSTRIA2010Q1.parameters
     initial_conditions = BeforeIT.AUSTRIA2010Q1.initial_conditions
 
-    parameters1 = deepcopy(parameters)
-    parameters2 = deepcopy(parameters)
-    initial_conditions1 = deepcopy(initial_conditions)
-    initial_conditions2 = deepcopy(initial_conditions)
-
-    model = BeforeIT.init_model(parameters1, initial_conditions1, T;)
-    data = BeforeIT.init_data(model)
-    for t in 1:(T - 1)
-        BeforeIT.run_one_epoch!(model; multi_threading = false)
-        BeforeIT.update_data!(data, model)
-    end
-
-
-    model2 = BeforeIT.init_model(parameters2, initial_conditions2, T;)
-    data2 = BeforeIT.init_data(model2)
-
-    for t in 1:(T - 1)
-        BeforeIT.run_one_epoch!(model2; multi_threading = false)
-        BeforeIT.update_data!(data2, model2)
+    function run_deterministic(parameters, initial_conditions, T, m)
+        model = BeforeIT.init_model(parameters, initial_conditions, T;)
+        data = BeforeIT.init_data(model)
+        for t in 1:(T - 1)
+            BeforeIT.run_one_epoch!(model; multi_threading = m)
+            BeforeIT.update_data!(data, model)
+        end
+        return model, data   
     end
 
+    model, data = run_deterministic(parameters, initial_conditions, T, false)
+    model2, data2 = run_deterministic(parameters, initial_conditions, T, false)
+    model3, data3 = run_deterministic(parameters, initial_conditions, T, true)
+    
     # loop over the data fields and compare them
     for field in fieldnames(typeof(data))
         @test isapprox(getproperty(data, field), getproperty(data2, field), rtol = 0.001)
+        @test isapprox(getproperty(data2, field), getproperty(data3, field), rtol = 0.001)
```

Comment:
@Tortar probably we can get to lower tolerances here can't we? (0.00001 should also work?)

Reply:
try before merging

Reply:
added 2 0s